build/android/patches/irrlicht-back_button.patch

```diff
@@ -1,14 +1,6 @@
---- irrlicht/source/Irrlicht/Android/CIrrDeviceAndroid.cpp	2014-06-03 20:56:21.289559503 +0200
-+++ irrlicht/source/Irrlicht/Android/CIrrDeviceAndroid.cpp.orig	2014-06-03 20:57:39.281556749 +0200
-@@ -423,6 +423,7 @@
- 			}
- 			
- 			device->postEventFromUser(event);
-+			status = 1;

```

Comment:
My explaining theory:

* As you can read from the definition of the `onInputEvent` function inside the `android_app` struct in `android_native_app_glue.h` (from the NDK), the function `onInputEvent` should "Return 1 if you have handled the event, 0 for any default dispatching".

* Irrlicht code sets the `onInputEvent` function pointer to the address of `CIrrDeviceAndroid::handleInput`

* While the docs of `CGUIEnvironment::postEventFromUser` don't reveal the meaning of the return value, the call of the [`onEvent`](http://irrlicht.sourceforge.net/docu/classirr_1_1_i_event_receiver.html) method inside the implementation reveals the common practice of returning `true` if the event was handled, and `false` if it wasn't handled.

So if you kept the state as is, every key, including volume down and up, won't be processed by the OS anymore. This probably causes #2122 .

If you went with @MoNTE48 's proposed solution, the OS would process all keys. It should only do that if minetest doesn't want to prevent it.